### ISSUE TITLE
model weight problem

### ISSUE THREAD

**Issue by liukc19** [FoundationVision/Groma#6]
I tried to finetune groma on REC dataset only(Refcoco/+/g), but get bad result on refcoco_val (with `iou@0.5 accu` and `m_iou` about 0.5). I also tried to evaluate groma on refcoco_val with `groma-7b-pretrain weight` and get the following result.

<img width="409" alt="截屏2024-05-09 17 28 32" src="https://github.com/FoundationVision/Groma/assets/76678452/030de1b1-6223-41bd-942c-df60c078f769">

Is this result normal?

**Comment by liukc19** [comment]
env:
+ 8 A100 40G
+ batch_size_per_device=8 
+ lr=5e-6

**Comment by liukc19** [comment]
I tried to export the pred_boxes and drew them in the pictures.
<img width="862" alt="截屏2024-05-10 09 46 50" src="https://github.com/FoundationVision/Groma/assets/76678452/42368efa-6361-4d9b-a976-80b8bec1644b">

<img width="1135" alt="截屏2024-05-10 09 44 28" src="https://github.com/FoundationVision/Groma/assets/76678452/6fd02e1a-d20b-4c43-a64d-7172e0309945">
The red bouding boxes are ground-truth boxes, and the blue boxes correspond to the prediction results and the model proposals respectively.
It seemed that the poor prediction results was caused by bad proposals.

As for why the loss decreased during the training process, I believe it was because the ground truth bounding boxes are injected.
<img width="826" alt="截屏2024-05-10 09 51 20" src="https://github.com/FoundationVision/Groma/assets/76678452/d97613a5-7166-4e6e-8dc4-da80f006c7ae">
Thanks for your help in advance.

**Comment by liukc19** [comment]
I also tried the weight of groma-7b-finetune, but got the same result. Is it possible that these errors come from this commit? #3 
Maybe the real weight is in this path `"vis_encoder_path": "checkpoints/dinov2-large"` ?

**Comment by machuofan** [comment]
Hi there, sorry for the late reply. I agree that the problem probably originates from model initialization. Could you please have a try by downloading the [DINOv2](https://huggingface.co/facebook/dinov2-large) checkpoint, and changing line [104-107](https://github.com/FoundationVision/Groma/blob/b7e23ed52639bc65cc4324a5e198dab5f9aabe67/groma/model/ddetr.py#L104-L107) in `groma/model/ddetr.py` from
```
if pretrained_vis_encoder is not None:
    self.vis_encoder = Dinov2Model.from_pretrained(pretrained_vis_encoder)
else:
    self.vis_encoder = Dinov2Model(config.vis_encoder_cfg)
``` 
to
```
self.vis_encoder = Dinov2Model.from_pretrained({path_to_dinov2_ckpt})
```
, which forces the model to load DINOv2 pretrain for initialization.

**Comment by liukc19** [comment]
thank u for ur suggestions, i'll try it later

**Comment by liukc19** [comment]
I tried the method u suggested but got the same result
![image](https://github.com/FoundationVision/Groma/assets/76678452/b4dba684-eec8-49a7-b89c-2dbfa1555742)
can you reproduce the result in your local environment(with finetuned model weight)?

**Comment by machuofan** [comment]
I found this error was caused by a mis-config of hyper parameter `nms_thres`, which was set to 0.0 but should be 0.6. It is now fixed in the latest commit. Please feel free to have a try.

**Comment by ryohachiuma** [comment]
Hi, @machuofan 

I tried to reproduce REC results on the RefCOCOg dataset.
I downloaded the model weights from [here](https://huggingface.co/FoundationVision/groma-7b-finetune) and followed the instructions at [here](https://github.com/FoundationVision/Groma/blob/main/docs/EVAL.md#rec-benchamrks) by running the following command.

```python torchrun --nnodes=1 --nproc_per_node=8 groma/eval/eval_rec.py     --model-name ./groma-7b-finetune     --img-prefix /workspace/dataset/MSCOCO/train2017     --ann-file ./rec_refcocog_val.json```

However, i got the results below. Am i missing sth?

```
rec_refcocog_val.json
iou@0.5 accu: 0.21956699346405228
m_iou: 0.20178459516537736
missing percentage: 0.0
```

Thanks in advance
